tests/bud_overlay_leaks.bats

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bats
+
+load helpers
+
+@test "bud overlay storage leaked mount" {
+  if test \! -e /usr/bin/fuse-overlayfs -a "$BUILDAH_ISOLATION" = "rootless"; then
+    skip "BUILDAH_ISOLATION = $BUILDAH_ISOLATION" and no /usr/bin/fuse-overlayfs present
+  fi
+
+  target=pull
+  run_buildah 125 --storage-driver=overlay bud --signature-policy ${TESTSDIR}/policy.json -t ${target} --pull-never ${TESTSDIR}/bud/pull
+  expect_output --substring "no such image"
+  run bash -c "mount | grep $TESTDIR"
+  [ "$status" -ne 0 ]
```

Comment:
I know this is normal BATS, but I really despise it: when a test fails in this manner, I find myself spending a lot of time trying to figure out what went wrong, why it's an error. Would you consider:
```bash
  leftover=$(mount | grep $TESTDIR)
  if [ -n "$leftover" ]; then
    die "buildah leaked a mount on error: $leftover"
  fi
```

Reply:
Sure